### PR TITLE
feat: experiment with alternative developer onboarding

### PR DESCRIPTION
**Pull request by matthewvolk**
## What/Why?
This PR contains documentation changes related to developer onboarding. 

Up until this point, the `create-catalyst` CLI has been the only path to get started with Catalyst as a developer. We've received feedback that some Catalyst developers are finding that working directly out of the monorepo makes it easier to checkout Catalyst integrations, and keep your Catalyst storefront up to date. We've been testing these patterns for a few weeks, and feel confident that this option is a path we should offer to the community in case they are interested in trying it out. Additionally, it consolidates the documentation overall; storefront developers, integration developers, and Catalyst contributors all have the same initial onboarding documentation.

It's important to note that there are still plenty of references to the `create-catalyst` CLI (on catalyst.dev, in the Catalyst repository README, etc.), and we expect many developers will default to using the CLI when creating a new storefront. We are OK with this, and will only remove references to the CLI if the community prefers the single-repository workflow.

## Testing
Preview: https://catalyst-dev-git-preview-bigcommerce-dx.vercel.app/docs/monorepo

Feel free to follow along with the documentation as you read.

### CODE REVIEW

File: docs/monorepo.md
Comment (chanceaclark) [inline docs/monorepo.md:1]: Seems like this should now be call `contributing.md`

File: docs/monorepo.md [hunk 1]
```diff
@@ -1,77 +1,323 @@
...
+## Overview
 
 The [main GitHub repository for Catalyst](https://github.com/bigcommerce/catalyst) is a [monorepo](https://en.wikipedia.org/wiki/Monorepo) with multiple codebases, including but not limited to the following:
 
-* The [core Next.js reference storefront application, in /core](https://github.com/bigcommerce/catalyst/tree/main/core)
-* [The component library, in /core/components/ui](https://github.com/bigcommerce/catalyst/tree/main/core/components/ui)
-* [The API client, in /packages/client](https://github.com/bigcommerce/catalyst/tree/main/packages/client)
-* [Product documentation, in /docs](https://github.com/bigcommerce/catalyst/tree/main/docs)
+- [The core Next.js reference storefront application, in `/core`](https://github.com/bigcommerce/catalyst/tree/main/core)
+- [Packages published to NPM, in `/packages`](https://github.com/bigcommerce/catalyst/tree/main/packages/client)
```
Comment (chanceaclark) [inline docs/monorepo.md:20]: This also includes private packages too, so maybe we could reword this?

File: docs/monorepo.md [hunk 2]
```diff
@@ -1,77 +1,323 @@
...
+- [Product documentation, in `/docs`](https://github.com/bigcommerce/catalyst/tree/main/docs)
 
-### The monorepo and the CLI
+It's important to note that we use several tools throughout the Catalyst monorepo, such as [Turborepo](https://turbo.build/repo/docs) and [pnpm Workspaces](https://pnpm.io/workspaces); you'll notice references to these tools as you spend more time in the repository, but most developers likely won't need to change any files, code, or configuration related to these tools in their normal Catalyst workflows.
 
-The [Catalyst CLI](https://www.npmjs.com/package/@bigcommerce/create-catalyst) installs just the Next.js application from `core/` onto your computer when you create a new Catalyst storefront.
+## Prerequisites
 
-If you are interested solely in building a headless storefront based on Catalyst, most of this monorepo is not relevant to you.
+1. **Node.js version `20.x`:** Catalyst currently supports Node.js version `20.x`, and does not support versions `<= 19` or `>= 21` ([click here to download Node.js](https://nodejs.org/en/download/package-manager))
```
Comment (chanceaclark) [inline docs/monorepo.md:27]: I don't think we should refer to odd node versions as they are not stable and can cause setup issues (that we've seen). I see that you mention we *don't* support them, but might just be best to omit it completely.

File: docs/monorepo.md [hunk 3]
```diff
@@ -1,77 +1,323 @@
...
+origin          git@github.com:<YOUR_GITHUB_USERNAME>/<YOUR_FORK_NAME>.git (push)
+upstream        git@github.com:bigcommerce/catalyst.git (fetch)
+upstream        git@github.com:bigcommerce/catalyst.git (push)
 ```
 
-You can find documentation for each field in the `.env.local` file, described in [.env.example](https://github.com/bigcommerce/catalyst/tree/main/.env.example).
+### 3. Enable Corepack
 
-4. If you use VS Code, use the following command to configure VSCode with the project-specific settings the Catalyst team has created:
+Enable Corepack so that you can use `pnpm` as your Node.js package manager with Catalyst. When working in the monorepo, you must use `pnpm`, you cannot use `npm` or `yarn`.
```
Comment (chanceaclark) [inline docs/monorepo.md:86]:
```suggestion
Enable Corepack so that you can use `pnpm` as your Node.js package manager with Catalyst. When working in the monorepo, you must use `pnpm`, **you cannot use `npm` or `yarn`**.
```

File: docs/monorepo.md [hunk 4]
```diff
@@ -1,77 +1,323 @@
...
+cp .vscode/settings.example.json .vscode/settings.json
 ```
 
-The `dev` script runs all packages and apps in watch mode.
+### 7. Run the development server!
+
+Finally, you can run the following command from the monorepo root to run the development server:
+
+```shell copy
+pnpm dev
```
Comment (chanceaclark) [inline docs/monorepo.md:132]: Let's include `run` as part of the commands as tons of external documentation use this. Shorthand is great but coming from `npm`, a lot of people might not know about it.
```suggestion
pnpm run dev
```

**Review by chanceaclark — COMMENTED** [review]
> We've received feedback that some Catalyst developers are finding that working directly out of the monorepo makes it easier to checkout Catalyst integrations, and keep your Catalyst storefront up to date.

I'd say that first part is an indication/small that the integration path needs some fine-tuning. I agree with the second part though.

---

Leaving a few comments for now. I see that below `### 8.` you have a few `@todo`'s.

File: docs/monorepo.md
Comment (matthewvolk) [reply docs/monorepo.md:1]: I was thinking it definitely needs a name change too - how do you feel about `developers.md` housed under `Getting-started`? I wanted to create a relatively shorter `non-developers.md` page that links out to non-technical resources and has some FAQ. That page would basically serve as an early-escape for non-technical folks who find themselves in the developer documentation. 

This would bifurcate the "Getting started" section into "Developers" and "Non-developers" really nicely IMO

Comment (chanceaclark) [reply docs/monorepo.md:1]: A lot of this document is around contributing to the repo and `contributing.md` is sort of a defacto standard. I'd prefer is we align with that instead.

File: docs/monorepo.md (hunk 4 above)
Comment (matthewvolk) [reply docs/monorepo.md:132]: Great point!

File: docs/monorepo.md
Comment (matthewvolk) [reply docs/monorepo.md:1]: I think I'm hesitant still; `contributing.md` is the defacto in SWE/OSS communities, 100% agree. However, it's pretty frequent that non-technical folks land on the catalyst.dev/docs page, and I'm not sure if that language would resonate with them. Instead, if they see a page called "Getting started > Developers" I think they'd be able to tell who that page is intended for. 

It would also duplicate with the root `CONTRIBUTING.md` file (though, that could be [solved with `title` overrides in Nextra](https://nextra.site/docs/guide/organize-files#_metajson))

I guess the real question is: Is the Catalyst.dev site intended for **all** merchants interested in Catalyst, or only developers? If the latter, where do non-technical merchants go for documentation?

File: docs/monorepo.md (hunk 1 above)
Comment (matthewvolk) [reply docs/monorepo.md:20]: QQ - it doesn't currently contain private packages right? You're just wanting to explain that a `packages/` folder in a monorepo can (generally speaking) contain private packages?

File: docs/monorepo.md
Comment (chanceaclark) [reply docs/monorepo.md:1]: Some websites have some sort contributing guide on them, take NextJS for example https://nextjs.org/docs/community/contribution-guide.

I see the content of this change has a ton of "contributing"-like references that would normally be on a `contributing.md` doc. And if we need to remove our root `CONTRIBUTING.md` we can always symlink it to this doc instead. You can change the title of this this page, but I don't necessarily see a need for that if you do want to make a separate page for non-developers. Most of the people looking at this doc though, will be a developer of some kind as I don't see a lot of non-devs setting up the repo from scratch.

Comment (jorgemoya) [reply docs/monorepo.md:1]: Would it be weird to include the suggestions of how to develop for your storefront in a `contributing.md` doc, though?

Comment (matthewvolk) [reply docs/monorepo.md:1]: Ah, nice - I like the idea of the symlink actually 💪 I just got out of a meeting with Georgia, she wanted to take a quick pass at the structure of the sidebar. I'll feed this to her, and then maybe we can discuss what the structure should look like in a future PR 👍

Comment (matthewvolk) [reply docs/monorepo.md:1]: > Would it be weird to include the suggestions of how to develop for your storefront in a `contributing.md` doc, though?

Great point - as I was writing this document, I had the opinion that in the future, a single document would be too crammed to fit all this information on. I see the future structure looking more like:

* Getting Started
  * Developers
    * Overview (`contributing.md` with just setup steps)
    * Storefront developers
    * Integration developers
  * Non-developers

Basically, eventually this single page will need to be broken out into multiple.

File: docs/monorepo.md [hunk 5]
```diff
@@ -1,77 +1,322 @@
...
+1. **Node.js version `20.x`:** Catalyst currently supports Node.js version `20.x` ([click here to download Node.js](https://nodejs.org/en/download/package-manager))
+2. **Corepack-enabled `pnpm`:** Node.js version `20.x` ships with Corepack by default; you should be able to run `corepack enable` from your terminal to meet this prerequisite (more information on the [official Node.js Corepack documentation](https://nodejs.org/docs/latest-v20.x/api/corepack.html))
+3. **Git version `2.35.x` or later:** ([Click here to download Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git))
+4. **A BigCommerce store or sandbox:** You can [create a trial store](https://bigcommerce.com/start-your-trial/) or [spin up a developer sandbox](https://start.bigcommerce.com/developer-sandbox/) to meet this prerequisite. If you already have a BigCommerce store, that will work fine too.
+5. **Permission to create API accounts on your store:** Our [documentation on High-Risk Permissions](https://support.bigcommerce.com/s/article/User-Permissions#highrisk) can help determine whether or not you have the correct permissions for the store you want to connect to.
```
Comment (christensenep) [inline docs/monorepo.md:31]: This feels clunky to me somehow, could we maybe put a little more detail here? Like maybe add something like"Ensure the store owner has granted you the `Create store-level API Accounts` permission." prior to the link. Something about the wording here makes me feel like the link will contain info on exactly which permissions you need, when that's not really the case.

**Review by christensenep — APPROVED** [review]
